BO: include Vue help for Symfony controllers (#989)
PrestaShop/docs: src/content/1.7/modules/concepts/templating/vuejs/faq.md

    ]);
```

> ### How to send data from Symfony modern controllers to Vue.js?

Reviewer: ```suggestion
### How to send data from Symfony controller to Vue.js?
```

PHP controller:

```php

> namespace Moduel\YorModule\Controller\Admin;

Reviewer: ```suggestion
namespace Module\YourModule\Controller\Admin;
```

use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;

class YourControllerNameController extends FrameworkBundleAdminController
{
    public function demoAction()
    {

>         return $this->render('@Modules/your-module/views/templates/admin/demo.html.twig', [

Reviewer: ```suggestion
        return $this->render('@Modules/yourmodule/views/templates/admin/demo.html.twig', [
```

            'yourModule' => [
                'keyA' => 'valueA',
                'keyB' => 'valueB',
            ]
        ]);
    }
}
```

Twig template:

```twig
{% extends '@PrestaShop/Admin/layout.html.twig' %}

> {% trans_default_domain "Module.YourModule.Admin.YourControllerName" %}

Reviewer: From what I remember we can't use more than 2 levels of domain in custom translation domains?

Can you confirm @eternoendless @sowbiba ?

{% block stylesheets %}

>   <link rel="stylesheet" href="https://ui-kit.prestashop.com/backoffice/latest/css/bootstrap-prestashop-ui-kit.css" type="text/css" media="all">

Reviewer: is it ok to give people example of loading external resource? I don't trust this path and PrestaShop servers, sorry :D

>   <link rel="stylesheet" href="{{ asset('../modules/your-module/views/css/app.css') }}" type="text/css" media="all">

Reviewer: ```suggestion
  <link rel="stylesheet" href="{{ asset('../modules/yourmodule/views/css/app.css') }}" type="text/css" media="all">
```

{% endblock %}

{% block content %}
  <div id="app"></div>
{% endblock %}

{% block javascripts %}
    {{ parent() }}

    <script>
      var yourModule = {{ yourModule|json_encode|raw }}
    </script>

>     <script src="https://ui-kit.prestashop.com/backoffice/latest/js/prestashop-ui-kit.js"></script>

Reviewer: same, I'd go for local path, we must have ui-kit globally with PrestaShop, right @NeOMakinG ?

Reviewer 2: Even if it's an example with external paths, it could lead to misunderstanding, as the UIKit is already included inside PS, maybe we could use another example? Maybe a dummy domain

Author: Sure. Let me change that.

Author: Done. I modified and included comments on Twig to clarify asset examples.

    <script src="{{ asset('../modules/your-module/views/js/chunk-vendors.js') }}" async defer></script>
    <script src="{{ asset('../modules/your-module/views/js/app.js') }}" async defer></script>
{% endblock %}
```

Use it in JavaScript, i.e. in your Vue Store:

```js

> const { yourModule } = window;
> const { keyA, keyB } = window.yourModule;

Reviewer 3: ```suggestion
const { yourModule } = window;
const { keyA, keyB } = yourModule;
```
 :trollface:

```php
namespace Module\YourModule\Controller\Admin;

use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;

class YourControllerNameController extends FrameworkBundleAdminController
{
    public function demoAction()
    {
        return $this->render('@Modules/yourmodule/views/templates/admin/demo.html.twig', [
            'yourModule' => [
                'keyA' => 'valueA',
                'keyB' => 'valueB',
            ]
        ]);
    }
}
```

Twig template:

```twig
{% extends '@PrestaShop/Admin/layout.html.twig' %}
{% trans_default_domain "Module.YourModule.Admin" %}

{% block stylesheets %}
  {# include some external stylesheets #}
  <link rel="stylesheet" href="https:/static.example.com/dummy/an-external-stylesheet-example.css" type="text/css" media="all">
  {# include some local stylesheets #}
  <link rel="stylesheet" href="{{ asset('../modules/yourmodule/views/css/app.css') }}" type="text/css" media="all">
{% endblock %}

{% block content %}
  <div id="app"></div>
{% endblock %}

{% block javascripts %}
    {{ parent() }}

    <script>
      var yourModule = {{ yourModule|json_encode|raw }}
    </script>

    {# include some external JavaScript #}

>     <script src="https:/static.example.com/dummy/an-external-javascript-example.js"></script>

Reviewer 3: ```suggestion
    <script src="https://static.example.com/dummy/an-external-javascript-example.js"></script>
```
